xarray/tests/test_interp.py

```diff
@@ -662,3 +662,68 @@ def test_datetime_interp_noerror():
         coords={"time": pd.date_range("01-01-2001", periods=50, freq="H")},
     )
     a.interp(x=xi, time=xi.time)  # should not raise an error
+
+
+def interp_1d(data, dim, vals):
+    selectors = {dim: vals}
+
+    old_coord = data[dim]
+
+    lower = data.sel(**selectors, method='ffill')
+    upper = data.sel(**selectors, method='bfill')
+    lower_x = data[dim].sel(**selectors, method='ffill')
+    upper_x = data[dim].sel(**selectors, method='bfill')
+    weight = np.abs(vals - lower_x)/np.abs(upper_x-lower_x)
```

Comment:
I would lean towards using the built in `abs()` here, which is slightly more generic than the NumPy function.